engine/runtime-instrument-common/src/main/java/org/enso/interpreter/instrument/IdExecutionService.java

```diff
@@ -237,9 +232,31 @@ public FunctionCallInfo(FunctionCallInstrumentationNode.FunctionCall call) {
         }
       }
 
-      notAppliedArguments = collectNotAppliedArguments(call);
+      return new FunctionPointer(moduleName, typeName, functionName);
     }
 
+    public static int[] collectNotAppliedArguments(Function function) {
+      FunctionSchema functionSchema = function.getSchema();
+      Object[] preAppliedArguments = function.getPreAppliedArguments();
```

Comment:
What if the `preAppliedArguments` array has zero size? Will the next line not throw an `ArrayIndexOutOfBoundsException`?